src/game/utils/batch/report.rs

```diff
@@ -0,0 +1,296 @@
+use std::{
+    fmt::Display,
+    fs::{create_dir, File},
+    io::Write,
+};
+
+use crate::{compute_average, game::utils::flop_helper::flop_to_string, Action, PostFlopGame};
+
+/// Returns the player's equity, EV, and EQR (in that order)
+/// Equity and EQR are float values from 0-100 (percentages)
+fn get_player_stats(game: &PostFlopGame, player: usize) -> (f32, f32, f32) {
+    let equity = game.equity(player);
+    let ev = game.expected_values(player);
+    let weights = game.normalized_weights(player);
+    let average_equity = compute_average(&equity, weights);
+    let average_ev = compute_average(&ev, weights);
+    (
+        100.0 * average_equity,
+        average_ev,
+        100.0 * average_ev / (average_equity * game.pot() as f32),
+    )
+}
+
+fn get_action_percentages(game: &PostFlopGame) -> Vec<f32> {
+    let player = game.current_player();
+    let cards = game.private_cards(player);
+    let strategy = game.strategy();
+    let actions = game.available_actions();
+    let weights = game.normalized_weights(player);
+
+    (0..actions.len())
+        .map(|i| compute_average(&strategy[i * cards.len()..(i + 1) * cards.len()], weights))
+        .collect()
+}
+
+// TODO is there any better way to do this?
+// I would rather not have to replay histories here b/c it is complicated and possibly slow
+// NOTE: Since this mutates the game, it un-caches weights
+fn get_action_evs(game: &mut PostFlopGame) -> Vec<f32> {
```

Comment:
@bkushigian Any ideas on a better way to do this? My concern is that all this resetting history and then replaying is a bit slow.

One option would be to compute action EVs on a second pass, recording the pot size at each node. This second pass would only require data from the first pass, and so would not need to replay any actions with `game`. However, this would significantly complicate the report generation code, and I'm not sure that it would be significantly faster.

Reply:
If I used `cf_calues`, I still need to get the weight for each hand. Is this possible without doing `game.play(i); game.cache_normalized_weights();`?

Reply:
Well, you need to call `play(...)` to get to the right place in the game tree. But once you're there, you have the  strategy and the weight at the current node, and that can be used to determine the weight of each hand after an action.

e.g., if I know I have AcKd with weight 0.2 and I know I bet it 50%, then I know that the bet node has it with weight 0.1.

But the more I think about it, the more I think actually playing and computing is the way to go, because this will be correct even in the presence of blocker effects like we talked about.

I'd want to compare the two approaches on narrow ranges, because this is where we should start to see some really big differences in EV.

BTW, there might be a way to "restore" the missing data instead of having to play the entire history again. Can we just copy weights and paste them back to memory/update the game's history directly?

Reply:
I'm now doing this in a better way: using the `expected_values_detail` directly, along with `strategy` and `normalized_weights`. This means we're no longer required to play the game at each action.

Reply:
Nice!